lib/node_modules/@stdlib/blas/base/cscal/README.md

```diff
@@ -0,0 +1,243 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2024 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# cscal
+
+> Scales a complex single-precision floating-point vector by a single-precision floating point constant.
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var cscal = require( '@stdlib/blas/base/cscal' );
+```
+
+#### cscal( N, cx, x, strideX )
+
+Scales values from `x` by `cx`.
+
+```javascript
+var Complex64Array = require( '@stdlib/array/complex64' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+
+var x = new Complex64Array( [ 1.0, 1.0, 1.0, 1.0, 1.0, 1.0 ] );
+var cx = new Complex64Array( [ 2.0, 0.0 ] );
+cscal( 3, cx, x, 1 );
+
+var z = x.get( 0 );
+// returns <Complex64>
+
+var re = realf( z );
+// returns 2.0
+
+var im = imagf( z );
+// returns 2.0
+```
+
+The function has the following parameters:
+
+-   **N**: number of indexed elements.
+-   **cx**: complex constant 
+-   **x**: input [`Complex64Array`][@stdlib/array/complex64].
+-   **strideX**: index increment for `x`.
+
+The `N` and stride parameters determine how values from `x` are scaled by `cx`. For example, to scale every other value in `x` by `cx`,
+
+```javascript
+var Complex64Array = require( '@stdlib/array/complex64' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+
+var x = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );
+var cx = new Complex64Array( [ 2.0, 0.0 ] );
+
+cscal( 2, cx, x, 2 );
+
+var z = x.get( 2 );
+// returns <Complex64>
+
+var re = realf( z );
+// returns 10.0
+
+var im = imagf( z );
+// returns 12.0
+```
+
+Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.
+
+<!-- eslint-disable stdlib/capitalized-comments -->
+
+```javascript
+var Complex64Array = require( '@stdlib/array/complex64' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+
+// Initial arrays...
+var x0 = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );
+var cx = new Complex64Array( [ 2.0, 2.0 ] );
+
+// Create offset views...
+var x1 = new Complex64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
+
+// Copy in reverse order every other value from `x1` into `y1`...
```

Comment:
This comment is incorrect. Either `x0` needs to grow and the stride be updated, or the comment needs to change.

Reply:
But also, the comment is just wrong more generally.